### ISSUE TITLE
Missing advapi32.dll on Windows UWP ARM64

### ISSUE THREAD

**Issue by camnewnham** [MichaCo/DnsClient.NET#149]
Continuation from https://github.com/sipsorcery-org/sipsorcery/issues/688

I've been using DnsClient via https://github.com/sipsorcery-org/sipsorcery and run into an interesting issue.
The usage here is in Unity via IL2CPP, targeting Windows Universal ARM64 (HoloLens 2).

Everything functions correctly on the target platform when using the .NET Framework build, but the .NET Standard build throws an exception originating at `NameServer.ResolveNameServer`, ultimately `System.DllNotFoundException: Unable to load DLL 'advapi32.dll': The specified module could not be found.`

```
System.AggregateException: One or more errors occurred. ---> System.InvalidOperationException: Error resolving name servers ---> System.DllNotFoundException: Unable to load DLL 'advapi32.dll': The specified module could not be found.
  at Microsoft.Win32.Win32RegistryApi.OpenSubKey (Microsoft.Win32.RegistryKey rkey, System.String keyName, System.Boolean writable) [0x00000] in <00000000000000000000000000000000>:0 
  at Microsoft.Win32.RegistryKey.OpenSubKey (System.String name, System.Boolean writable) [0x00000] in <00000000000000000000000000000000>:0 
  at Microsoft.Win32.RegistryKey.OpenSubKey (System.String name) [0x00000] in <00000000000000000000000000000000>:0 
  at DnsClient.Windows.NameResolutionPolicy.Resolve (System.Boolean includeGenericServers, System.Boolean includeDirectAccessServers) [0x00000] in <00000000000000000000000000000000>:0 
  at DnsClient.NameServer.ResolveNameResolutionPolicyServers () [0x00000] in <00000000000000000000000000000000>:0 
  at DnsClient.NameServer.ResolveNameServers (System.Boolean skipIPv6SiteLocal, System.Boolean fallbackToGooglePublicDns) [0x00000] in <00000000000000000000000000000000>:0 
  at DnsClient.LookupClient..ctor (DnsClient.LookupClientOptions options, DnsClient.DnsMessageHandler udpHandler, DnsClient.DnsMessageHandler tcpHandler) [0x00000] in <00000000000000000000000000000000>:0 
  at DnsClient.LookupClient..ctor (DnsClient.LookupClientOptions options) [0x00000] in <00000000000000000000000000000000>:0 
  at DnsClient.LookupClient..ctor () [0x00000] in <00000000000000000000000000000000>:0 
  at SIPSorcery.Net.RtpIceChannel.StartGathering () [0x00000] in <00000000000000000000000000000000>:0 
```

I'm not particularly familiar with the Windows APIs in question, but thoughts would be:
- The dll doesn't exist or can't be accessed on Windows ARM64
- `RuntimeInformation.IsOSPlatform(OSPlatform.Windows)` returns true and assumes the functions are available.
- `Registry.LocalMachine.OpenSubKey` subsequently throws a `DllNotFoundException`

Related issues (unable to access advapi.dll on this Windows flavour):
- https://stackoverflow.com/questions/69481411/how-to-install-syswow64-dlls-files-in-hololens
- https://github.com/yukihiko/ThreeDPoseUnitySample/issues/4
Related (unable to access registry on UWP)
- https://stackoverflow.com/questions/48899692/how-to-access-registry-key-in-a-uwp-app
---

A possible solution could be to [check the proccessor architecture](https://stackoverflow.com/a/54538014) if running on windows to determine if running on ARM/ARM64.

Alternatively just putting the calls that require this library (OpenSubKey) in a try/catch and reverting to a fallback code path could work?

Any thoughts? Happy to do some further investigation/testing.

**Comment by camnewnham** [comment]
This PR implements the second solution, simply wrapping `Registry.LocalMachine.OpenSubKey` in a `try/catch` for `DllNotFoundException`, and resolves the issue in my scenario.

**Comment by MichaCo** [comment]
Hi @camnewnham,
Thanks for reporting the issue, that's a bit unfortunate how that Windows registry Nuget doesn't handle errors by itself... 

I'll have a look into that. Thanks for the PR, too btw. Might take a bit until that gets released though

**Comment by MichaCo** [comment]
@Cheesebaron & @camnewnham, this should be fixed in 1.6.1 which just released.
